### ISSUE TITLE
Style/Sample: safe autocorrect changes the result when a seeded random: is passed

### ISSUE THREAD

**Issue by corsonknowles** [rubocop/rubocop#15576]
## Expected behavior

`Style/Sample`'s autocorrection is classified **safe** (the cop has no `@safety` section and `extend AutoCorrector`), so running `rubocop -a` should not change what the code returns.

## Actual behavior

For a `shuffle` given an explicitly seeded `random:`, the corrected code selects **different elements**. The cop's documentation lists this shape as `# bad`, so it is flagged and corrected deliberately:

```ruby
#   [1, 2, 3].shuffle(random: Random.new).first
```

Deterministic sampling from a fixed seed is the common reason to pass `random:` at all, so the correction silently changes the sampled set.

## Steps to reproduce

`sample_repro.rb`:

```ruby
# frozen_string_literal: true
ITEMS = (1..20).to_a

def deterministic_subset
  ITEMS.shuffle(random: Random.new(123_456)).first(5)
end
```

```console
$ rubocop --only Style/Sample sample_repro.rb
sample_repro.rb:5:9: C: [Correctable] Style/Sample: Use sample(5, random: Random.new(123_456)) instead of shuffle(random: Random.new(123_456)).first(5).

$ rubocop --only Style/Sample -a sample_repro.rb     # safe autocorrect
```

Result: `ITEMS.sample(5, random: Random.new(123_456))`

The two expressions do not agree:

```ruby
items = (1..20).to_a
items.shuffle(random: Random.new(123_456)).first(5)  # => [20, 14, 13, 10, 6]
items.sample(5, random: Random.new(123_456))         # => [2, 12, 20, 14, 19]
```

The single-element form differs too:

```ruby
items.shuffle(random: Random.new(9)).first  # => 4
items.sample(random: Random.new(9))         # => 2
```

Both consume the generator differently, so any fixed seed produces a different selection.

## Why this matters in practice

I hit this on a real codebase where two files sample a deterministic subset of records to audit, seeded so the subset is reproducible across runs. `rubocop -a` would have quietly changed which records get audited — no failure, no diff in behaviour that a test would notice unless it asserts on the specific subset.

## Suggested fix

Skip the autocorrection when a `random:` argument is present, while still registering the offense. `shuffle(random: r).first` → `sample(random: r)` is only an equivalence when the caller does not depend on the generator's sequence, and the cop cannot know that.

A narrower version is possible — a bare `Random.new` is distribution-equivalent, whereas `Random.new(seed)` is not, and those are distinguishable syntactically — but declining to correct whenever `random:` is passed is simpler and covers the case where the generator arrives in a variable.

Alternatively, adding an `@safety` section and marking the correction unsafe would at least stop `-a` from applying it.

## RuboCop version

```console
$ rubocop --version
1.89.0
$ ruby -v
ruby 3.4.10 (2026-06-30 revision 2b0b7728dc) +PRISM [arm64-darwin25]
```
